core/shared/src/main/scala-2/org/scalacheck/package.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * ScalaCheck
+ * Copyright (c) 2007-2021 Rickard Nilsson. All rights reserved.
+ * http://www.scalacheck.org
+ *
+ * This software is released under the terms of the Revised BSD License.
+ * There is NO WARRANTY. See the file LICENSE for the full text.
+ */
+
+package org.scalacheck
+
+object `package` {
+  implicit class `by-name label :|`(private val prop: Prop) extends AnyVal {
+
+    /** Label this property.
+      *
+      * The label is evaluated lazily. The operator name is chosen for its precedence btween boolean operators and
+      * others.
+      */
+    def =|=(label: => String): Prop = prop.map(_.label(label))
```

Comment:
Another part of my PR (#979) was to avoid evaluating the label if it will never be displayed to the user. What do you think about doing the same here?

```scala
def =|=(label: => String): Prop =
  prop.map(r =>
    r.status match {
      case Prop.False | Prop.Exception(_) => r.label(l)
      case Prop.Proof | Prop.True | Prop.Undecided => r
    })
```